packages/eslint-plugin-react-hooks/src/ExhaustiveDeps.js

```diff
@@ -93,6 +93,28 @@ export default {
       const reactiveHook = node.callee;
       const reactiveHookName = getNodeWithoutReactNamespace(reactiveHook).name;
       const declaredDependenciesNode = node.arguments[callbackIndex + 1];
+      const isEffect = /Effect($|[^a-z])/g.test(reactiveHookName);
+
+      // Check the declared dependencies for this reactive hook. If there is no
+      // second argument then the reactive callback will re-run on every render.
+      // So no need to check for dependency inclusion.
+      if (!declaredDependenciesNode && !isEffect) {
+        // These are only used for optimization.
+        if (
+          reactiveHookName === 'useMemo' ||
+          reactiveHookName === 'useCallback'
+        ) {
+          // TODO: Can this have a suggestion?
+          reportProblem({
+            node: reactiveHook,
+            message:
+              `React Hook ${reactiveHookName} does nothing when called with ` +
+              `only one argument. Did you forget to pass an array of ` +
+              `dependencies?`,
+          });
+        }
```

Comment:
What about "else" here? Do we ignore such hooks?

Reply:
Rest would be effects, right? For effects we do allow to not specify deps.

Reply:
> Rest would be effects, right? For effects we do allow to not specify deps.

Seeing this code for the first time today, it was quite hard to read through and reason about its control flow (especially due to lots of early returns); maybe you're right and it's fine.

Reply:
It's a pretty complex rule. :-) I'd happily take refactorings provided they don't break tests.